src/storage/DomainDecomposition.cpp

```diff
@@ -799,114 +1354,554 @@ void DomainDecomposition::doGhostCommunication(bool sizesFirst, bool realToGhost
+                                        recvSizes[i - gsize[k]]);
+                                }
+                                LOG4ESPP_DEBUG(logger, "exchanging ghost cell sizes done");
+                            }
+                            sendSizes.clear();
+                            recvSizes.clear();
+                        }
+
+                        // prepare send and receive buffers
+                        longint receiver, sender;
+                        outBuffer.reset();
+                        // inBuffer.reset();
+
+                        if ((new_dir > 0 ? new_dir : -new_dir) == 2)
+                        {
+                            if (realToGhosts)
+                            {
+                                receiver = cnode[k];
+                                if (k == 0) sender = nodeGrid.getNodeNeighborIndex(oppositeDir);
+                                // absolute x-positions (counts in cells) of current real&ghost
+                                // cells
+                                int cell_x1, cell_x2, itmp;
+                                real cx_flag = (shift[2] > .0 ? 1.0 : -1.0);
+                                for (int i = rsize[k], end = rsize[k + 1]; i < end; ++i)
+                                {
+                                    cell_x1 = nodeGrid.getNodePosition(0) * getInt3DCellGrid()[0] +
+                                              (commCells[dir].reals[i] - getFirstCell()) %
+                                                  (getInt3DCellGrid()[0] + 2) -
+                                              1;
+                                    cell_x2 =
+                                        cnode[k] % getInt3DNodeGrid()[0] * getInt3DCellGrid()[0] +
+                                        (commCells[dir].ghosts[i] - getFirstCell()) %
+                                            (getInt3DCellGrid()[0] + 2) -
+                                        1;
+                                    itmp = static_cast<int>(
+                                        floor(((cell_x1 - cell_x2) * cellGrid.getCellSize(0) +
+                                               cx_flag * offs) /
+                                                  Lx +
+                                              0.5));
+                                    packPositionsEtc_LEBC(outBuffer, *commCells[dir].reals[i],
+                                                          extradata, shift,
+                                                          cx_flag * offs - (itmp + .0) * Lx);
+                                }
+
+                                // exchange particles, odd-even rule
+                                if (nodeGrid.getNodePosition(coord) % 2 == 0)
+                                {
+                                    outBuffer.send(receiver, DD_COMM_TAG);
+                                    if (k == 0) inBuffer.recv(sender, DD_COMM_TAG);
```

Comment:
@xzhh there is a warning here:
```
/__w/espressopp/espressopp/src/storage/DomainDecomposition.cpp: In member function 'virtual void espressopp::storage::DomainDecomposition::doGhostCommunication(bool, bool, int)':
/__w/espressopp/espressopp/src/storage/DomainDecomposition.cpp:1672:62: error: 'sender' may be used uninitialized [-Werror=maybe-uninitialized]
 1672 |                                     if (k == 0) inBuffer.recv(sender, DD_COMM_TAG);
      |                                                 ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/__w/espressopp/espressopp/src/storage/DomainDecomposition.cpp:1633:43: note: 'sender' was declared here
 1633 |                         longint receiver, sender;
      |                                           ^~~~~~
cc1plus: all warnings being treated as errors
```

Can you look into that?

Reply:
Updated with the new commit. The compiler does not recognize that the condition (if k==0) is the same when sender/receiver is initialized and when it is first called.